### PR TITLE
add a new resource cce_node_attach

### LINKED ISSUE #1178: Add ability to add/remove CCE nodes
I'd like to add support to remove nodes from CCE cluster but keeping ECS/BMS instances. Also should be able to add ECS/BMS instances to CCE cluster.

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

add a new resource cce_node_attach

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1178 

**Special notes for your reviewer**:

**Release note**:

```release-note
Creating this resource can add a cce node to a cluster from an existing ecs server.
Deleting will remove the previously added node. Note that the ecs server won't be deleted.
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud' TESTARGS='-run TestAccCCENodeAttachV3_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run TestAccCCENodeAttachV3_basic -timeout 360m -parallel 4
=== RUN   TestAccCCENodeAttachV3_basic
=== PAUSE TestAccCCENodeAttachV3_basic
=== CONT  TestAccCCENodeAttachV3_basic
--- PASS: TestAccCCENodeAttachV3_basic (819.26s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       819.328s
```

### CODE REVIEW

File: docs/resources/cce_node_attach.md [hunk 1]
```diff
@@ -0,0 +1,150 @@
+---
+subcategory: "Cloud Container Engine (CCE)"
+---
+
+# huaweicloud\_cce\_node\_attach
```
Comment (ShiChangkuo) [inline docs/resources/cce_node_attach.md]: **\\** is unnecessary.

File: docs/resources/cce_node_attach.md [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+Add a node from an existing ecs server to a CCE cluster.
+
+## Basic Usage
```
Comment (ShiChangkuo) [inline docs/resources/cce_node_attach.md:8]: suggest to add the full example into [examples/cce](https://github.com/huaweicloud/terraform-provider-huaweicloud/tree/master/examples)
Here, we just keep `huaweicloud_cce_node_attach`, and use variables to set the cluster_id and server_id

File: docs/resources/cce_node_attach.md [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+
+## Argument Reference
+The following arguments are supported:
+
+* `cluster_id` - (Required, String, ForceNew) Specifies the ID of the cluster.
+  Changing this parameter will create a new resource.
+
+* `name` - (Optional, String) Specifies the Node Name.
+
+* `server_id` - (Required, String, ForceNew) Specifies the flavor id.
```
Comment (ShiChangkuo) [inline docs/resources/cce_node_attach.md]: server id or flavor id ?

File: docs/resources/cce_node_attach.md [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+* `preinstall` - (Optional, String, ForceNew) Specifies the script required before installation.
+  The input value can be a Base64 encoded string or not.
+  Changing this parameter will create a new resource.
+
+* `postinstall` - (Optional, String, ForceNew) Specifies the script required after installation.
+  The input value can be a Base64 encoded string or not.
+  Changing this parameter will create a new resource.
+
+* `image_id` - (Optional, String, ForceNew) Specifies the image ID. Changing this parameter will create a new resource.
```
Comment (ShiChangkuo) [inline docs/resources/cce_node_attach.md]: what's the difference between `os` and `image_id`?

File: huaweicloud/resource_huaweicloud_cce_node_attach.go [hunk 5]
```diff
@@ -0,0 +1,382 @@
...
+		Update: resourceCCENodeV3Update,
+		Delete: resourceCCENodeAttachV3Delete,
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(20 * time.Minute),
+			Delete: schema.DefaultTimeout(20 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"cluster_id": {
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_cce_node_attach.go:33]: mising **region**

File: huaweicloud/resource_huaweicloud_cce_node_attach.go [hunk 6]
```diff
@@ -0,0 +1,382 @@
...
+				Optional: true,
+				ForceNew: true,
+				Default:  "[{\"queue\":4}]",
+			},
+			"nic_threshold": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Default:  "0.3:0.6",
+			},
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_cce_node_attach.go:94]: defined but not added in the web doc.

File: huaweicloud/resource_huaweicloud_cce_node_attach.go [hunk 7]
```diff
@@ -0,0 +1,382 @@
...
+			"runtime": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"ecs_group_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			//(node/ecs_tags)
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_cce_node_attach.go]: the comment is misleading

File: huaweicloud/resource_huaweicloud_cce_node_attach.go [hunk 8]
```diff
@@ -0,0 +1,382 @@
...
+		return fmtp.Errorf("Error adding HuaweiCloud Node: %s", err)
+	}
+
+	nodeID, err := getResourceIDFromJob(nodeClient, s.JobID, "CreateNode", "InstallNode")
+	if err != nil {
+		return err
+	}
+	d.SetId(nodeID)
+
+	// logp.Printf("[DEBUG] Waiting for CCE Node (%s) to become available", s.Metadata.Name)
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_cce_node_attach.go]: don't keep the debug message

File: docs/resources/cce_node_attach.md (hunk 1 above)
Comment (Jason-Zhang9309) [reply docs/resources/cce_node_attach.md]: fixed

File: docs/resources/cce_node_attach.md (hunk 3 above)
Comment (Jason-Zhang9309) [reply docs/resources/cce_node_attach.md]: fixed

File: huaweicloud/resource_huaweicloud_cce_node_attach.go (hunk 5 above)
Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_cce_node_attach.go:27]: This resource is a relation type resource. It won't create any instance. So region will not work.

File: huaweicloud/resource_huaweicloud_cce_node_attach.go (hunk 6 above)
Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_cce_node_attach.go:94]: Due to the API problem, these two parameters have to be contained in request body, but they won't work acturally.
They will work while adding a BMS node which is not supported for now.

File: huaweicloud/resource_huaweicloud_cce_node_attach.go (hunk 7 above)
Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_cce_node_attach.go]: fixed

File: huaweicloud/resource_huaweicloud_cce_node_attach.go (hunk 8 above)
Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_cce_node_attach.go]: fixed

File: docs/resources/cce_node_attach.md (hunk 4 above)
Comment (Jason-Zhang9309) [reply docs/resources/cce_node_attach.md]: This parameter specifies a custom image ID while adding a BMS node. It won't work as Adding a BMS node is not supported for now. Remove is from the doc could be better

File: docs/resources/cce_node_attach.md (hunk 2 above)
Comment (Jason-Zhang9309) [reply docs/resources/cce_node_attach.md:8]: fixed

File: huaweicloud/resource_huaweicloud_cce_node_attach.go (hunk 5 above)
Comment (Jason-Zhang9309) [reply huaweicloud/resource_huaweicloud_cce_node_attach.go:33]: fixed

**Pull request merged**